### ISSUE TITLE
Confusion between Message and MessageTemplate

### ISSUE THREAD

**Issue by piotrmaslanka** [tintoy/seqlog#79]
According to Seq docs, keys sent as Message should be immediately human-readable, while MessageTemplates might contain placeholders, such as {device_id} or {0}.

Right now we're sending a fully formatted message as a MessageTemplate.

That's a bug as I see it.

**Comment by tintoy** [comment]
I agree - providing they are using named-hole formatting (which we know at configuration time).

**Comment by webJose** [comment]
I agree as well!  Hello.

This bug directly affects a cool Seq functionality.  Allow me to provide a visual:

<img width="1193" height="469" alt="Image" src="https://github.com/user-attachments/assets/47e8a520-d1b8-450e-b2be-d16e18adfe74" />

The screenshot comes from the Seq user interface, Events page.  Item 1shows the hash calculated by seq on `@mt`, which this library is currently sending incorrectly:  Instead of sending the message template, which in this case was `Total repositories found: {RepositoryCount}`, is sending the formatted message, which is the result of replacing values in `@mt` with the values the log record contains (item 2 in this case), which should end up being `@m`.

The consequence is that Item 3 in the screenshot doesn't work as desired:  Instead of finding all records whose message template was the same, it only finds all instances where the actual formatted message ended up identical.

Do let me know if I can be of help.  Thanks!

**Comment by tintoy** [comment]
Yes, this should be handled by `StructuredLogger` and `SeqLogHandler`, but it sounds like there’s a bug somewhere. I’m hoping to look into this issue as well, this weekend.

**Comment by tintoy** [comment]
https://github.com/tintoy/seqlog/blob/373dbe565c6020040e08cb8f98d992a2c20d5f4c/seqlog/structured_logging.py#L527
and
https://github.com/tintoy/seqlog/blob/373dbe565c6020040e08cb8f98d992a2c20d5f4c/seqlog/structured_logging.py#L585
and
https://github.com/tintoy/seqlog/blob/373dbe565c6020040e08cb8f98d992a2c20d5f4c/seqlog/structured_logging.py#L124
and
https://github.com/tintoy/seqlog/blob/373dbe565c6020040e08cb8f98d992a2c20d5f4c/seqlog/structured_logging.py#L158

(`msg` field of `StructuredLogRecord` definitely _should_ be the message template, and it does look like that is what’s passed to seq as `_mt_`, so it may be that  the calling code is not using `StructuredLogger`)

https://github.com/tintoy/seqlog/blob/373dbe565c6020040e08cb8f98d992a2c20d5f4c/seqlog/structured_logging.py#L190

**Comment by tintoy** [comment]
Hmm, that looks correct to me. Can you try passing `use_clef=True‘ to `seqlog.configure_from_file()`?

**Comment by webJose** [comment]
With `use_clef=True` I get:  `404 Client Error: Not Found for url: http://localhost:5341/ingest/clef`

My guess is I need a newer Seq?  I'll check this out.

**Comment by webJose** [comment]
Ok, updated my Docker image of Seq to the latest version.  The issue is gone and now logs post with `use_clef=True`.  The problem remains:

```json
{
	"@t": "2025-07-26T02:10:42.8804500Z",
	"@mt": "Total repositories found for topic web-worker: 290",
	"@m": "Total repositories found for topic web-worker: 290",
	"@i": "44fa1676",
	"AppCode": "CG",
	"AppName": "Curious GHeorge",
	"LoggerName": "__main__",
	"RepositoryCount": 290,
	"ThreadId": 23372,
	"ThreadName": "MainThread",
	"Topic": "web-worker",
	"TransactionId": "99cff5e1-c11a-4950-a4cc-2311b980f8fa"
}
```
`@m == @mt` 🥹

**Comment by tintoy** [comment]
Hmm - I’ll have to try this myself, again; last time I tried it, message template was definitely being used. I wonder if it’s the use of a formatter that’s causing problems? Will take a look tomorrow morning when I get home :)

**Comment by tintoy** [comment]
@piotrmaslanka @webJose - I _think_ that this is the actual cause of the behaviour you were seeing:

https://github.com/tintoy/seqlog/commit/7c38baf6fb46b87a3d7999b9a86aedd71662be0c

I've published [v0.4.2](https://pypi.org/project/seqlog/0.4.2/) so if you get a chance, would you mind trying it out to see if it works for you?

**Comment by webJose** [comment]
Sir, yes, sir!

I just finished testing:

<img width="2713" height="440" alt="Image" src="https://github.com/user-attachments/assets/ab4ff418-c023-4f22-ae07-bcc26645201e" />

Great job!  Do you see how the Seq interface colorizes the values?  That's because it now knows how the template looks like.

Here's the exported JSON for one of those entries:

```json
{
	"@t": "2025-07-27T04:03:24.8293280Z",
	"@mt": "Total repositories found for topic {Topic}: {RepositoryCount}",
	"@m": "Total repositories found for topic python: 526152",
	"@i": "91913a91",
	"AppCode": "CG",
	"AppName": "Curious GHeorge",
	"LoggerName": "__main__",
	"RepositoryCount": 526152,
	"ThreadId": 47392,
	"ThreadName": "MainThread",
	"Topic": "python",
	"TransactionId": "5dfbb40e-d74f-4fc1-b463-05abe3690bbe"
}
```

Success!

Thank you very much for your support.  You're the man!
